### PR TITLE
[query] simplify fs interface

### PR DESCRIPTION
**Pull request by cseed**
It was too tightly coupled to the Hadoop interface (still is).  Remove fs.{FilePath, FileSystem}.

### CODE REVIEW

**Comment by cseed** [comment]
Pushed an additional change: Delete specialized open/create (read/write) functions.  Just use `using` plus open/create.

File: hail/src/main/scala/is/hail/expr/ir/TableIR.scala [hunk 1]
```diff
@@ -14,6 +14,7 @@ import is.hail.linalg.{BlockMatrix, BlockMatrixMetadata, BlockMatrixReadRowBlock
 import is.hail.rvd._
 import is.hail.sparkextras.ContextRDD
 import is.hail.utils._
+import org.apache.hadoop.fs.{FSDataInputStream, FSDataOutputStream}
```
Comment (akotlar) [inline hail/src/main/scala/is/hail/expr/ir/TableIR.scala]: not needed

File: hail/src/main/scala/is/hail/io/IndexBTree.scala [hunk 2]
```diff
@@ -1,10 +1,11 @@
 package is.hail.io
 
-import java.io.Closeable
+import java.io.{Closeable, DataOutputStream}
 import java.util.Arrays
 
 import is.hail.utils._
 import is.hail.io.fs.FS
+import org.apache.hadoop.fs.FSDataInputStream
```
Comment (akotlar) [inline hail/src/main/scala/is/hail/io/IndexBTree.scala]: not needed

**Comment by akotlar** [comment]
minor import thing

**Pull request merged**